x/delayedack/eibc.go

```diff
@@ -28,25 +57,19 @@ func (im IBCMiddleware) eIBCDemandOrderHandler(ctx sdk.Context, rollappPacket co
 
 	switch t := rollappPacket.Type; t {
 	case commontypes.RollappPacket_ON_RECV:
-		// Handle eibc demand order if exists - Start by validating the memo
-		memo := make(map[string]interface{})
-		err := json.Unmarshal([]byte(data.Memo), &memo)
-		if err != nil || memo[eibcMemoObjectName] == nil {
-			logger.Debug("Memo is empty or failed to unmarshal", "memo", data.Memo)
-			return nil
-		}
-
-		// Currently not supporting eibc with PFM: https://github.com/dymensionxyz/dymension/issues/599
-		if memo[PFMMemoObjectName] != nil {
-			err = fmt.Errorf("EIBC packet with PFM is currently not supported")
+		var err error
+		packetMetaData, err = parsePacketMetadata(data.Memo)
+		if errors.Is(err, ErrMemoIsPFM) {
 			return err
 		}
-		// Unmarshal the packet metadata from the memo
-		err = json.Unmarshal([]byte(data.Memo), packetMetaData)
-		if err != nil {
-			logger.Error("parsing packet metadata from memo", "error", err)
+		if errors.Is(err, ErrMemoEibcEmpty) || errors.Is(err, ErrMemoUnmarshal) {
+			logger.Debug("skipping demand order creation", "error", err)
 			return nil
 		}
+		if err != nil {
+			return err
+		}
+		return err
```

Comment:
this DOES slightly change the logging, because we no longer do an .Error log
but I think that's fine?